setup.ps1

```diff
@@ -0,0 +1,287 @@
+    }
+    else {
+      Write-Log "Virtual environment created."
+    }
+  }
+
+  # Activate the virtual environment and check if it was successful
+  Write-Log "Virtual environment found. Activating virtual environment..."
+  & $ActivateVenv 2>&1 | Out-File $LogFilePath -Append
+  # Check if virtual environment is activated
+  if ($env:VIRTUAL_ENV) {
+    Write-Host "Virtual environment is activated at: $($env:VIRTUAL_ENV)"
+  }
+  else {
+    Write-Log "Failed to activate virtual environment." -Level 'ERROR'
+    Exit-Script -exitCode 1
+  }
+  
+  # Ensure pip
+  python -m ensurepip --default-pip 2>&1 | Out-File $LogFilePath -Append
+
+  # Pull latest updates only if the repository state is not dirty
+  Write-Log "Checking if the repository is clean..."
+  $Status = & "C:\Program Files\Git\cmd\git.exe" status --porcelain
+  if ($Status) {
+    Write-Log "Repository is dirty. Skipping pull."
+  }
+  else {
+    Write-Log "Pulling latest updates..."
+    & "C:\Program Files\Git\cmd\git.exe" pull 2>&1 | Out-File $LogFilePath -Append
+    if ($LASTEXITCODE -ne 0) {
+      Write-Log "Failed to pull updates from Git." -Level 'ERROR'
+      Exit-Script -exitCode 1
+    }
+  }
+
+  if (-not (Test-Path "$VenvDir\Lib\site-packages\talib")) {
+    # Install TA-Lib using the virtual environment's pip
+    Write-Log "Installing TA-Lib using virtual environment's pip..."
+    python -m pip install --find-links=build_helpers\ --prefer-binary TA-Lib 2>&1 | Out-File $LogFilePath -Append
+    if ($LASTEXITCODE -ne 0) {
+      Write-Log "Failed to install TA-Lib." -Level 'ERROR'
+      Exit-Script -exitCode 1
+    }
+  }
+
+  # Present options for requirement files
+  $SelectedIndices = Get-UserSelection -prompt "Select which requirement files to install:" -options $RequirementFiles -defaultChoice 'A'
+
+  # Cache the selected requirement files
+  $SelectedRequirementFiles = @()
+  $PipInstallArguments = @()
+  foreach ($Index in $SelectedIndices) {
+    $RelativePath = $RequirementFiles[$Index]
+    if (Test-Path $RelativePath) {
+      $SelectedRequirementFiles += $RelativePath
+      $PipInstallArguments += "-r", $RelativePath  # Add each flag and path as separate elements
+    }
+    else {
+      Write-Log "Requirement file not found: $RelativePath" -Level 'ERROR'
+      Exit-Script -exitCode 1
+    }
+  }
+  if ($PipInstallArguments.Count -ne 0) {
+    & pip install @PipInstallArguments # Use array splatting to pass arguments correctly
+  }
+
+  # Install freqtrade from setup using the virtual environment's Python
+  Write-Log "Installing freqtrade from setup..."
+  pip install -e . 2>&1 | Out-File $LogFilePath -Append
+  if ($LASTEXITCODE -ne 0) {
+    Write-Log "Failed to install freqtrade." -Level 'ERROR'
+    Exit-Script -exitCode 1
+  }
+
+  $UiOptions = @("Yes", "No")
+  $InstallUi = Get-UserSelection -prompt "Do you want to install the freqtrade UI?" -options $UiOptions -defaultChoice 'B' -allowMultipleSelections $false
```

Comment:
I'm not entirely sure if i'd present "this" as an option.

we don't for the linux/macos script - and this is a pretty lightweight install (it's just downloading a bunch of files) - so i'm not entirely sure if i'd bother the user with it, or just install it automatically.


at any rate, we should be consistent with naming, and use `freqUI` here (as well as below).

Reply:
Ok np, I'll fix that.

Reply:
I would be fine with automatically installing it too

Reply:
I think i'd prefer that, yeah - that way it's aligned.

Reply:
Ok, I am about to make it auto installing.